hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2260,10 +2258,8 @@ private class Emit(
 
         new NDArrayEmitter(mb, nDims, shapeArray,
           xP.shape.pType, xP.elementType, setup, ndt.setup, ndt.m) {
-          override def outputElement(idxVars: Array[Code[Long]]): Code[_] = {
-            val elementLocation = xP.getElementAddress(idxVars, ndAddress, mb)
-            Region.loadIRIntermediate(outputElementPType)(elementLocation)
-          }
+          override def outputElement(idxVars: Array[Code[Long]]): Code[_] =
+            xP.loadElementToIRIntermediate(idxVars, ndAddress, mb)
```

Comment:
This may not work, not sure yet why outputElementPType is different from data.pType

Reply:
Oh I see; this is fine, and also a somewhat confusing way to access xP.elementType. Do we need the xP.elementType on NDArrayEmitter?

Reply:
I use the `outputElementPType` field in the `emitLoops` method of `NDArrayEmitter`. That's how I know the type of the NDArray that gets build in the end. You could argue that a more reasonable thing to do would be to just have the `NDArrayEmitter` store the type of the NDArray you'd like to get out in the end, especially if we get to a world where there are many NDArray PTypes.